genie-common/src/main/java/com/netflix/genie/common/util/ExponentialBackOff.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.common.util;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Implementation of AdaptiveBackOff that increases the delay exponentially and optionally resets in case of
+ * success and/or failure.
+ *
+ * @author mprimi
+ * @since 3.3.8
+ */
+@ThreadSafe
+public class ExponentialBackOff implements AdaptiveBackOff {
+    private final long minInterval;
+    private final long maxInterval;
+    private final float factor;
+    private final boolean resetOnSuccess;
+    private final boolean resetOnFailure;
+    private long currentInterval;
+    private long nextExecutionTime;
+
+    public ExponentialBackOff(
+        final long minInterval,
+        final long maxInterval,
+        final float factor,
+        final boolean resetOnSuccess,
+        final boolean resetOnFailure
+    ) {
+        this.minInterval = minInterval;
+        this.maxInterval = maxInterval;
+        this.factor = factor;
+        this.resetOnSuccess = resetOnSuccess;
+        this.resetOnFailure = resetOnFailure;
+        this.currentInterval = minInterval;
+
+        if (maxInterval < minInterval) {
+            throw new IllegalArgumentException("Maximum interval cannot be smaller than minimum");
+        }
+
+        if (factor < 1) {
```

Comment:
This case (if you want) could be covered with `@Min` `javax.validation`. However then you'd have to validate this implementation not sure if you want to do that or not just pointing it out.